sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1040,3 +1040,44 @@ def test_init():
                                max_iter=1, random_state=random_state).fit(X)
 
         assert gmm2.lower_bound_ >= gmm1.lower_bound_
+
+
+def test_gaussian_mixture_setting_best_params():
+    """`GaussianMixture`'s best_parameters, `n_iter_` and `lower_bound_`
+    must be set appropriately in the case of divergence.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/18216
+    """
+    rnd = np.random.RandomState(0)
+    n_samples = 30
+    X = rnd.uniform(size=(n_samples, 3))
+
+    # specify initialization parameters that won't lead to a convergence
+    means_init = np.array([
```

Comment:
How were they found?
I'm asking because once the bug is fixed, it will be harder to reproduce the failure.

Reply:
Hello @lorentzenchr ! Thank you for your comments! I made some changes accordingly. 

As for the initialization parameters, they were the intermediary results I got when fitting [GaussianMixtureIC](https://github.com/scikit-learn/scikit-learn/pull/19562), an implementation we are proposing to automatically select the best Gaussian mixture model based on BIC/AIC. The steps of calculating those parameters as in GaussianMixtureIC include fitting `AgglomerativeClustering`,  running `OneHotEncoder`, and computing Gaussian parameters from the onehot. I didn't want to include them all in this test so added the parameters directly. Please let me know if you think there is a better way of specifying them.

Reply:
Could you then do 2 things:

1. Just add a comment that those initialization parameters were found to lead to non-convergence.
2. Check for non-convergence, i.e.
```python
# check that the fit did non converge
assert not gmm.converged_
```